### PR TITLE
Feat: analyst can create facility/organisation

### PR DESCRIPTION
**Pull request by dleard**
- Moves add facility/organisation into the analyst dashboard
- Splits AddOrganisationFacility file into two modals
- Creates new AddFacility / AddOrganisation pages & fragment containers

### CODE REVIEW

File: app/components/AddOrganisationModal.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  onAddOrganisation?: (...args: any[]) => void;
+}
+
+const AddOrganisationModal: React.FunctionComponent<Props> = (props) => {
+  const {onAddOrganisation} = props;
+  const [isModalVisible, setModalVisible] = useState(false);
+
+  const customFormats = {
+    'postal-code': /[a-z]\d[a-z]\s?\d[a-z]\d/i,
+    duns: /^\d{9}$/
```
Comment (matthieu-foucault) [inline app/components/AddOrganisationModal.tsx:18]: The application form does not collect duns number anymore. So we may want something else in this view. CRA number maybe? Maybe we can get a quick clarification on what to do here (or is that a leftover from a previous version of the form)

File: app/containers/Facilities/AddFacility.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  const selectFacility = (name, id, operatorName, facilityType, bcghgid) => {
+    props.handleFacilityChange(id);
+    props.handleInputChange('');
+    setSelectedFacilityDetails({name, id, operatorName, facilityType, bcghgid});
+  };
+
+  const handleAddFacility = async (variables) => {
+    const {environment} = props.relay;
+    const response = await createFacilityMutation(environment, variables);
+    console.log(response);
```
Comment (matthieu-foucault) [inline app/containers/Facilities/AddFacility.tsx:38]: toast message?

File: app/containers/Facilities/AddFacilityModal.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  onAddFacility?: (...args: any[]) => void;
+}
+
+const AddFacilityModal: React.FunctionComponent<Props> = (props) => {
+  const {onAddFacility, query} = props;
+  const [isModalVisible, setModalVisible] = useState(false);
+
+  const customFormats = {
+    'postal-code': /[a-z]\d[a-z]\s?\d[a-z]\d/i,
+    duns: /^\d{9}$/
```
Comment (matthieu-foucault) [inline app/containers/Facilities/AddFacilityModal.tsx:24]: no duns here

File: app/containers/Facilities/OrganisationRowIdField.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import {FieldProps} from 'react-jsonschema-form';
+import {createFragmentContainer, graphql} from 'react-relay';
+import {OrganisationRowIdField_query} from 'OrganisationRowIdField_query.graphql';
+
+interface Props extends FieldProps<number> {
+  query: OrganisationRowIdField_query;
+}
+
+/**
+ * Fetches the list of fuels and their row ids
```
Comment (matthieu-foucault) [inline app/containers/Facilities/OrganisationRowIdField.tsx:11]:
```suggestion
 * Fetches the list of organisations and their row ids
```

File: app/components/AddOrganisationModal.tsx (hunk 1 above)
Comment (dleard) [reply app/components/AddOrganisationModal.tsx:18]: yeah I did a lot of copy/paste moving stuff around from existing code, this is probably just a leftover

**Pull request merged**